From 82e37721b57cf96efe812a6aa831316f75632e38 Mon Sep 17 00:00:00 2001
From: Molly Miller <mm@flyingcircus.io>
Date: Tue, 1 Jul 2025 16:11:41 +0200
Subject: [PATCH 3/4] netlink: don't manage kernel neighbour table entries for
 L2-VNI

From experience, zebra's tracking of MAC-IP mappings for SVI
interfaces in L2 VPNs doesn't always stay in sync with the kernel,
which can lead to it advertising stuck type-2 routes with mappings for
IP addresses which are no longer locally assigned.

To hedge against this on the receiving side, let's stop populating the
neighbour table with mappings learned from EVPN -- but only for L2
VPNs. For L3 VPNs, the neighbour table is used for resolving the
nexthop MAC address for IP routes inside the VPN, so we can't stub out
neighbour table management on the data plane side entirely.

In L2 VPNs, the kernel will fall back to dynamic neighbour discovery
with ARP and NDP.
---
 zebra/zebra_evpn_neigh.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/zebra/zebra_evpn_neigh.c b/zebra/zebra_evpn_neigh.c
index a2fcd77fe0..439f94c690 100644
--- a/zebra/zebra_evpn_neigh.c
+++ b/zebra/zebra_evpn_neigh.c
@@ -159,7 +159,7 @@ int zebra_evpn_rem_neigh_install(struct zebra_evpn *zevpn,
 		flags |= DPLANE_NTF_ROUTER;
 	ZEBRA_NEIGH_SET_ACTIVE(n);
 
-	dplane_rem_neigh_add(vlan_if, &n->ip, &n->emac, flags, was_static);
+	/* dplane_rem_neigh_add(vlan_if, &n->ip, &n->emac, flags, was_static); */
 
 	return ret;
 }
@@ -824,7 +824,7 @@ static int zebra_evpn_neigh_uninstall(struct zebra_evpn *zevpn,
 	ZEBRA_NEIGH_SET_INACTIVE(n);
 	n->loc_seq = 0;
 
-	dplane_rem_neigh_delete(vlan_if, &n->ip);
+	/* dplane_rem_neigh_delete(vlan_if, &n->ip); */
 
 	return 0;
 }
-- 
2.50.1 (Apple Git-155)

